# Adding microsatellites to big table

In [1]:
from pandas import DataFrame
import pandas as pd

In [2]:
ls ../panopea_data/data-results/Geoduck-transcriptome_v3_bigtable_v08.csv

../panopea_data/data-results/Geoduck-transcriptome_v3_bigtable_v08.csv*


In [3]:
bigv8 = pd.read_csv ("../panopea_data/data-results/Geoduck-transcriptome_v3_bigtable_v08.csv",engine='python')#, delimiter=" ", 
#                        index_col=None, names= encabezado, header= None)
bigv8


ContigID  length  GC_content   CpG_o/e  male_unique_count  \
0       comp137246_c1_seq26    1458     45.4047  0.719212                  0   
1        comp142515_c0_seq2    5369     38.8527  0.524659               3714   
2        comp116682_c1_seq2     701     45.6491  0.525637                 31   
3        comp141240_c0_seq5    4445     41.5973  0.517579                  3   
4        comp127090_c0_seq1    3402     39.3592  0.979904               1435   
5        comp124055_c0_seq3    6718     46.0107  0.725103                 41   
6        comp142437_c1_seq3    2126     39.6049  0.408168                118   
7        comp143196_c0_seq1    2564     41.5757  0.541800               1180   
8        comp143573_c0_seq1    4514     43.0882  0.517212                794   
9        comp142424_c2_seq1    3060     46.4379  0.644757                 15   
10       comp140491_c1_seq2    1424     43.7500  0.660264                 71   
11       comp133082_c0_seq2    2412     34.0796  0.646259                  0   
12       comp142337_c2_seq1    2177     42.0763  0.471202                452   
13       comp140336_c1_seq1    1365     35.1648  0.554706                134   
14       comp144262_c1_seq8    4863     39.4612  0.521609                 35   
15       comp140963_c2_seq2    4207     43.9981  0.540484                  0   
16       comp123995_c0_seq1    1364     40.8358  0.617201                630   
17       comp139776_c0_seq1    2731     38.3010  0.450660                 56   
18       comp138448_c0_seq6    2327     45.1225  0.670407                  0   
19       comp142670_c2_seq1    1303     36.8381  0.673410                 82   
20       comp136546_c0_seq2    2998     45.5971  0.638410                 41   
21       comp138450_c0_seq1    2303     41.6848  0.443395                 76   
22       comp140253_c0_seq2    2956     34.0325  0.399531                609   
23       comp143626_c0_seq1    4168     37.8599  0.499395                626   
24       comp137128_c0_seq1    1812     38.7417  0.443910                 14   
25       comp123534_c0_seq1    1276     45.3762  0.342953                 72   
26       comp141720_c0_seq4    1916     39.8747  0.557889                  6   
27       comp143076_c0_seq6    3762     40.4572  0.626853                  1   
28       comp135897_c0_seq2     703     42.8165  0.320759                  0   
29       comp142444_c0_seq2    2207     38.0607  0.814511                732   
...                     ...     ...         ...       ...                ...   
153952    comp99949_c0_seq1     513     28.2651  0.412522                  4   
153953    comp99951_c0_seq1     367     40.0545  0.486776                  6   
153954    comp99951_c1_seq1     239     29.2887  0.196082                  5   
153955    comp99952_c0_seq1     229     45.4148  0.343418                 24   
153956    comp99952_c0_seq2     249     44.9799  0.320518                  0   
153957    comp99956_c0_seq1     221     30.7692  1.168445                  4   
153958    comp99957_c0_seq1     414     38.6473  0.470448                 22   
153959    comp99964_c0_seq1     339     37.4631  0.592045                 12   
153960    comp99965_c0_seq1     225     28.4444  0.000000                  8   
153961    comp99966_c0_seq1     383     28.1984  0.263738                 38   
153962    comp99967_c0_seq1     222     28.8288  0.871111                  0   
153963    comp99970_c0_seq1     474     29.9578  0.378111                 26   
153964    comp99971_c0_seq1     265     30.9434  0.449331                 19   
153965    comp99972_c0_seq1     484     44.2149  0.171606                  2   
153966    comp99972_c0_seq2     717     40.7252  0.204880                  7   
153967    comp99973_c0_seq1     280     60.7143  1.246136                  3   
153968    comp99974_c0_seq1     226     29.2035  0.450406                  5   
153969    comp99979_c0_seq1     219     32.8767  0.865478                  3   
153970    comp99980

there are 153982 rows

In [4]:
ls ../panopea_data/data-results/geoduck-v3-msat/geoduck-v3microsatellites.csv

../panopea_data/data-results/geoduck-v3-msat/geoduck-v3microsatellites.csv*


In [5]:
micros = pd.read_csv ("../panopea_data/data-results/geoduck-v3-msat/geoduck-v3microsatellites.csv",
                     engine='python')
micros

ContigID  records_id    id motif  type  start   end  \
0     comp100188_c0_seq1         143     0    AG     2    189   209   
1     comp100330_c0_seq1         236     1    AT     2    119   135   
2     comp100888_c0_seq1         618     2    AC     2    293   313   
3     comp101080_c0_seq1         769     3    AG     2      1    19   
4     comp101629_c0_seq1        1167     4    AT     2    296   312   
5     comp101629_c0_seq2        1168     5    AT     2    559   575   
6     comp102145_c0_seq1        1560     6    AT     2     23    39   
7     comp102949_c0_seq1        2161     7    AT     2      1    19   
8     comp103695_c1_seq1        2715     8    AT     2    190   208   
9     comp103744_c0_seq1        2749     9    AT     2     34    52   
10    comp103954_c0_seq1        2893    10    AT     2    199   215   
11    comp103954_c0_seq2        2894    11    AT     2    221   237   
12    comp104465_c2_seq1        3275    12    AT     2      0    16   
13    comp104873_c0_seq1        3589    13    AC     2      1    19   
14    comp104946_c0_seq1        3650    14    AT     2    114   130   
15    comp105344_c0_seq1        3919    15    AG     2    153   169   
16    comp105417_c0_seq1        3961    16    AT     2    302   318   
17    comp105874_c0_seq3        4315    17    AC     2      1    19   
18    comp105923_c0_seq1        4358    18    AG     2    218   234   
19    comp105939_c0_seq1        4369    19    AT     2     47    63   
20    comp106069_c0_seq1        4498    20    AT     2     72    88   
21    comp106069_c0_seq2        4499    21    AT     2    956   972   
22    comp106553_c0_seq1        4984    22    AC     2    281   301   
23    comp107101_c1_seq1        5518    23    AC     2      2    20   
24    comp107101_c1_seq2        5519    24    AC     2     84   100   
25    comp107161_c0_seq1        5561    25    AT     2      0    16   
26    comp107637_c0_seq1        6002    26    AT     2     54    70   
27    comp107843_c0_seq1        6220    27    AT     2    318   334   
28    comp107943_c0_seq1        6308    28    AT     2    260   278   
29    comp107981_c0_seq1        6339    29    AC     2     42    58   
...                  ...         ...   ...   ...   ...    ...   ...   
1336   comp89008_c0_seq1      146453  1336    AC     2      1    21   
1337   comp89276_c0_seq2      146628  1337    AT     2    191   209   
1338   comp89906_c0_seq1      147007  1338    AT     2     72    90   
1339   comp90437_c1_seq1      147378  1339    AT     2   3090  3106   
1340   comp90869_c0_seq1      147673  1340    AT     2      0    18   
1341   comp91181_c0_seq1      147885  1341    AT     2    326   342   
1342   comp91376_c1_seq1      148022  1342    AT     2      0    18   
1343   comp91644_c0_seq1      148183  1343    AC     2    763   779   
1344   comp91871_c0_seq1      148334  1344    AG     2    182   202   
1345   comp91871_c0_seq2      148335  1345    AG     2    283   303   
1346   comp93274_c0_seq1      149273  1346    AT     2    224   240   
1347   comp93940_c0_seq2      149711  1347    AT     2      0    18   
1348   comp95100_c0_seq1      150546  1348    AC     2      1    23   
1349   comp95105_c0_seq1      150548  1349    AT     2      1    19   
1350   comp95235_c0_seq1      150627  1350    AC     2    266   286   
1351   comp95994_c0_seq1      151138  1351    AT     2      1    17   
1352    comp9690_c0_seq1      151773  1352    AC     2      1    21   
1353   comp96987_c0_seq1      151819  1353    AT     2    104   122   
1354   comp97380_c0_seq1      152129  1354    AC     2     80    96   
1355   comp97380_c0_seq1      152129  1355    AC     2    291   311   
1356   comp97380_c0_seq2      152130  1356    AC     2     84   100   
1357   comp97380_c0_seq2      152130  1357    AC     2    295   315   
1358   comp97863_c0_seq1      152451  1358    AT     2    700   718   
1359   comp98468_c0_seq1      152835  1359    AC     2      0    20   
1360   comp98654_c0_seq1      152982  13

there are 1366 micros

## merging databases 

In [6]:
bigmicros = pd.merge(bigv8, micros, on= "ContigID", how="outer")
bigmicros

ContigID  length  GC_content   CpG_o/e  male_unique_count  \
0       comp137246_c1_seq26    1458     45.4047  0.719212                  0   
1        comp142515_c0_seq2    5369     38.8527  0.524659               3714   
2        comp116682_c1_seq2     701     45.6491  0.525637                 31   
3        comp141240_c0_seq5    4445     41.5973  0.517579                  3   
4        comp127090_c0_seq1    3402     39.3592  0.979904               1435   
5        comp124055_c0_seq3    6718     46.0107  0.725103                 41   
6        comp142437_c1_seq3    2126     39.6049  0.408168                118   
7        comp143196_c0_seq1    2564     41.5757  0.541800               1180   
8        comp143573_c0_seq1    4514     43.0882  0.517212                794   
9        comp142424_c2_seq1    3060     46.4379  0.644757                 15   
10       comp140491_c1_seq2    1424     43.7500  0.660264                 71   
11       comp133082_c0_seq2    2412     34.0796  0.646259                  0   
12       comp142337_c2_seq1    2177     42.0763  0.471202                452   
13       comp140336_c1_seq1    1365     35.1648  0.554706                134   
14       comp144262_c1_seq8    4863     39.4612  0.521609                 35   
15       comp140963_c2_seq2    4207     43.9981  0.540484                  0   
16       comp123995_c0_seq1    1364     40.8358  0.617201                630   
17       comp139776_c0_seq1    2731     38.3010  0.450660                 56   
18       comp138448_c0_seq6    2327     45.1225  0.670407                  0   
19       comp142670_c2_seq1    1303     36.8381  0.673410                 82   
20       comp136546_c0_seq2    2998     45.5971  0.638410                 41   
21       comp138450_c0_seq1    2303     41.6848  0.443395                 76   
22       comp140253_c0_seq2    2956     34.0325  0.399531                609   
23       comp143626_c0_seq1    4168     37.8599  0.499395                626   
24       comp137128_c0_seq1    1812     38.7417  0.443910                 14   
25       comp123534_c0_seq1    1276     45.3762  0.342953                 72   
26       comp141720_c0_seq4    1916     39.8747  0.557889                  6   
27       comp143076_c0_seq6    3762     40.4572  0.626853                  1   
28       comp135897_c0_seq2     703     42.8165  0.320759                  0   
29       comp142444_c0_seq2    2207     38.0607  0.814511                732   
...                     ...     ...         ...       ...                ...   
153977    comp99949_c0_seq1     513     28.2651  0.412522                  4   
153978    comp99951_c0_seq1     367     40.0545  0.486776                  6   
153979    comp99951_c1_seq1     239     29.2887  0.196082                  5   
153980    comp99952_c0_seq1     229     45.4148  0.343418                 24   
153981    comp99952_c0_seq2     249     44.9799  0.320518                  0   
153982    comp99956_c0_seq1     221     30.7692  1.168445                  4   
153983    comp99957_c0_seq1     414     38.6473  0.470448                 22   
153984    comp99964_c0_seq1     339     37.4631  0.592045                 12   
153985    comp99965_c0_seq1     225     28.4444  0.000000                  8   
153986    comp99966_c0_seq1     383     28.1984  0.263738                 38   
153987    comp99967_c0_seq1     222     28.8288  0.871111                  0   
153988    comp99970_c0_seq1     474     29.9578  0.378111                 26   
153989    comp99971_c0_seq1     265     30.9434  0.449331                 19   
153990    comp99972_c0_seq1     484     44.2149  0.171606                  2   
153991    comp99972_c0_seq2     717     40.7252  0.204880                  7   
153992    comp99973_c0_seq1     280     60.7143  1.246136                  3   
153993    comp99974_c0_seq1     226     29.2035  0.450406                  5   
153994    comp99979_c0_seq1     219     32.8767  0.865478                  3   
153995    comp99980

there are 154007 rows
### There are more sequences than before!!
### let's try mergin with microsatellite only 

In [7]:
bigmicrosin = pd.merge(bigv8, micros, on= "ContigID", how="inner")
bigmicrosin

ContigID  length  GC_content   CpG_o/e  male_unique_count  \
0      comp131711_c0_seq2     983     37.1312  0.860193                 24   
1      comp141289_c0_seq6    1467     34.6285  0.413580                 24   
2      comp135170_c0_seq1    1648     35.4976  0.251797                652   
3      comp110143_c0_seq1     921     32.1390  0.651499                442   
4      comp139838_c2_seq4    1579     40.6586  0.661371                  2   
5      comp132023_c0_seq7     852     36.9718  0.517682                  5   
6      comp143738_c0_seq1    3703     34.9446  0.426903               1159   
7      comp143363_c0_seq1    3250     45.6308  0.516718                 46   
8      comp143363_c0_seq2    3284     45.8892  0.557885                 62   
9      comp142951_c1_seq2    1115     36.8610  0.373334                 89   
10     comp120618_c1_seq1    1004     34.4622  0.558858                 43   
11     comp133661_c0_seq1    3196     41.6771  0.376674                378   
12     comp141169_c0_seq2    1390     34.3165  0.984744                  9   
13     comp139838_c2_seq3    1555     40.4502  0.676816                  8   
14     comp137819_c0_seq1    1244     35.0482  1.129360               1892   
15     comp138666_c0_seq1    2346     34.3990  0.580862               2206   
16     comp134005_c0_seq7    1664     40.7452  1.303149               2765   
17     comp143950_c0_seq2    5041     41.5195  0.578297                108   
18     comp144313_c1_seq1    6499     48.2382  0.433545               1556   
19     comp139040_c3_seq1    1708     33.4309  0.468932                 87   
20     comp142283_c0_seq2    3577     38.3841  0.505420                486   
21     comp135890_c0_seq2    1449     41.4079  0.890948                 46   
22     comp124054_c0_seq1     711     30.2391  0.828205               3384   
23     comp144348_c0_seq1   14086     47.6501  0.435083               3408   
24     comp134777_c1_seq1    1167     43.8732  0.971670                  3   
25    comp140817_c1_seq17    4349     44.9069  0.699020                  0   
26     comp116360_c0_seq1     886     37.1332  0.434248                416   
27     comp130412_c0_seq2    1350     33.1852  0.358283                  0   
28     comp140730_c0_seq2    3569     38.9745  0.465855                122   
29     comp140730_c0_seq4    3283     39.0192  0.400324                 78   
...                   ...     ...         ...       ...                ...   
1336    comp89008_c0_seq1     331     30.8157  0.256175                 11   
1337    comp89276_c0_seq2     209     28.7081  0.753606                  4   
1338    comp89906_c0_seq1     296     29.7297  0.768642                  4   
1339    comp90437_c1_seq1    3143     31.7849  0.702048               2833   
1340    comp90869_c0_seq1     318     28.9308  0.457680                 28   
1341    comp91181_c0_seq1     342     32.4561  0.335401                  4   
1342    comp91376_c1_seq1     309     31.3916  1.080151                  1   
1343    comp91644_c0_seq1     925     30.5946  0.556547               1403   
1344    comp91871_c0_seq1     223     45.2915  0.000000                  7   
1345    comp91871_c0_seq2     324     37.0370  0.211659                 19   
1346    comp93274_c0_seq1     262     37.7863  1.225865                  4   
1347    comp93940_c0_seq2     705     32.6241  0.482241                 14   
1348    comp95100_c0_seq1     210     44.7619  0.769739                 89   
1349    comp95105_c0_seq1     404     36.3861  0.375280                 31   
1350    comp95235_c0_seq1     287     37.2822  0.000000                 62   
1351    comp95994_c0_seq1     489     29.8569  0.184038                 10   
1352     comp9690_c0_seq1     217     35.0230  0.906464                  5   
1353    comp96987_c0_seq1     204     35.7843  0.330121                  8   
1354    comp97380_c0_seq1     320     36.2500  0.417565                  0   
1355    comp97380_c0_seq1     320     3

There are 1366 rows √

## grouping for motif of the microsatellites

In [8]:
mp = bigmicrosin.groupby("motif")["ContigID"].count()
mp

motif
AAAC        3
AAACT       1
AAAG        2
AAAT        1
AACAGC      2
AACC        3
AACGC       1
AAG         2
AATC        1
AATG        3
AATT        2
AC        350
ACAG        1
ACC         1
ACGG        1
AG        244
AGAT        2
AGC         2
AT        735
ATC         8
CG          1
Name: ContigID, dtype: int64

## to obtain those annotated contigs with microsatellites

In [9]:
df = bigmicrosin[bigmicrosin["ProteinName"].notnull()]
df

ContigID  length  GC_content   CpG_o/e  male_unique_count  \
0     comp131711_c0_seq2     983     37.1312  0.860193                 24   
1     comp141289_c0_seq6    1467     34.6285  0.413580                 24   
2     comp135170_c0_seq1    1648     35.4976  0.251797                652   
3     comp110143_c0_seq1     921     32.1390  0.651499                442   
4     comp139838_c2_seq4    1579     40.6586  0.661371                  2   
5     comp132023_c0_seq7     852     36.9718  0.517682                  5   
6     comp143738_c0_seq1    3703     34.9446  0.426903               1159   
7     comp143363_c0_seq1    3250     45.6308  0.516718                 46   
8     comp143363_c0_seq2    3284     45.8892  0.557885                 62   
9     comp142951_c1_seq2    1115     36.8610  0.373334                 89   
10    comp120618_c1_seq1    1004     34.4622  0.558858                 43   
11    comp133661_c0_seq1    3196     41.6771  0.376674                378   
12    comp141169_c0_seq2    1390     34.3165  0.984744                  9   
13    comp139838_c2_seq3    1555     40.4502  0.676816                  8   
14    comp137819_c0_seq1    1244     35.0482  1.129360               1892   
15    comp138666_c0_seq1    2346     34.3990  0.580862               2206   
16    comp134005_c0_seq7    1664     40.7452  1.303149               2765   
17    comp143950_c0_seq2    5041     41.5195  0.578297                108   
18    comp144313_c1_seq1    6499     48.2382  0.433545               1556   
19    comp139040_c3_seq1    1708     33.4309  0.468932                 87   
20    comp142283_c0_seq2    3577     38.3841  0.505420                486   
21    comp135890_c0_seq2    1449     41.4079  0.890948                 46   
22    comp124054_c0_seq1     711     30.2391  0.828205               3384   
23    comp144348_c0_seq1   14086     47.6501  0.435083               3408   
24    comp134777_c1_seq1    1167     43.8732  0.971670                  3   
25   comp140817_c1_seq17    4349     44.9069  0.699020                  0   
26    comp116360_c0_seq1     886     37.1332  0.434248                416   
27    comp130412_c0_seq2    1350     33.1852  0.358283                  0   
28    comp140730_c0_seq2    3569     38.9745  0.465855                122   
29    comp140730_c0_seq4    3283     39.0192  0.400324                 78   
..                   ...     ...         ...       ...                ...   
100   comp141819_c0_seq1    1424     36.8680  0.381322                285   
101   comp139878_c0_seq1    2139     37.1669  0.819924               2790   
102   comp135323_c0_seq1    3472     32.7189  0.418892               4051   
103   comp143077_c1_seq4    2459     40.5856  0.565350                443   
104   comp128513_c0_seq3    2891     39.6057  0.547066                  1   
105   comp129066_c0_seq1     874     33.5240  0.329847                 40   
106   comp140272_c0_seq1    2814     41.8266  0.472959                  2   
107   comp134778_c2_seq1    1458     43.1413  0.591770                  6   
108   comp139098_c0_seq1    1731     41.2478  1.002244                111   
109   comp139707_c2_seq1    1297     46.4148  0.932060                  5   
110   comp130067_c0_seq7    1409     32.8602  0.507579              17284   
111   comp142358_c1_seq4    3056     32.9843  0.513223                295   
112   comp137176_c0_seq1    1266     44.2338  0.890860                  9   
113   comp137250_c1_seq1     758     36.2797  0.969144                  5   
114   comp137250_c1_seq2     742     35.9838  0.962762                  6   
115   comp135117_c0_seq4    1258     38.3148  0.517413                  4   
116  comp144300_c0_seq13    3569     37.9938  0.637383                 20   
117  comp144300_c0_seq22    2300     35.3913  0.611229                  0   
118   comp144300_c0_seq3    3533     38.0696  0.633656                 15   
119   comp139071_c1_seq5    2034     41.9371  0.627119                  4   
120   comp13

## there are 130 annotated microsatellites

## grouping by motif of the microsatellite

In [10]:
mp = df.groupby("motif")["ContigID"].count()
mp

motif
AAG     2
AC     24
ACC     1
AG     24
AT     78
ATC     1
Name: ContigID, dtype: int64

## grouping by sex-based expression

In [11]:
mp2 = df.groupby("sex-based_expresssion")["ContigID"].count()
mp2

sex-based_expresssion
female         2
female>male    1
Name: ContigID, dtype: int64

### there are only three microsatellites associated with over-expressed genes in females

In [12]:
for row in df.index:
    row1 = df.ix[row]
    if str(row1["sex-based_expresssion"])!="nan":
        print (row1["ContigID"], row1["sex-based_expresssion"] , str(row1["ProteinName"])[:60])

comp140811_c0_seq2 female Oxalate:formate antiporter (OFA) (Oxalate:formate antiport p
comp137592_c0_seq4 female Photoreceptor-specific nuclear receptor (Nuclear receptor su
comp137176_c0_seq1 female>male Transmembrane protein 178B
